01-basics/20-weather/WeatherApp.js

```diff
@@ -3,49 +3,61 @@ import { getWeatherData, WeatherConditionIcons } from './weather.service.ts'
 
 export default defineComponent({
   name: 'WeatherApp',
+  setup() {
+    const weatherData = getWeatherData()
+    const weatherConditionIcons = WeatherConditionIcons;
 
-  template: `
-    <div>
-      <h1 class="title">Погода в Средиземье</h1>
+    const isDayTime = (current, sunrise, sunset) => {
+      return new Date(`1970-01-01T${current}:00`).getTime() > new Date(`1970-01-01T${sunrise}:00`).getTime() && new Date(`1970-01-01T${current}:00`).getTime() < new Date(`1970-01-01T${sunset}:00`).getTime();
+    }
 
-      <ul class="weather-list unstyled-list">
-        <li class="weather-card weather-card--night">
-          <div class="weather-alert">
+    return {
+      weatherData, weatherConditionIcons, isDayTime
+    }
+  },
+
+  template:
+    `<div v-for="(weather, index) in weatherData" key="weather.geographic_name">
```

Comment:
Without `v-bind:` (or just `:`) it is a plain string `"weather.geographic_name"` and cannot be used as a `key`.

Reply:
Oops, without helping of editor it is possible to make such mistake. Installed Vue Inline Template :)